### PR TITLE
Update User Guide and Admin links to point to new resources.

### PR DESCRIPTION
**Pull request by PaulJuliusMartinez**
When browsing documentation, clicking on links and seeing pages like [this](https://kubernetes.io/docs/user-guide/replication-controller/) is a pretty frustrating user experience.

I wrote a script (`update-user-guide-links.py`) that finds all the files that include the content `{% include user-guide-content-moved.md %}`, then parses the content below to find where the files redirects to. It then executes a `sed` command to replace all links to the old file with links to the new file in all the relevant Markdown files. (I am unsure if other files need to be updated or other build scripts need to be re-run to update other files. It seems like there are a lot of html files laying around that might need to be updated too.)

`update-user-guide-links.py` should be run from the `docs/` directory via `python update-user-guide-links.py`. (Script works with Python 2.7.)

`docs/api.md` has two possible redirect points, so those links are not updated.

Note that `sed` adds a lot of missing new-lines at the ends of files. I ran this command to not add the white-space changes after running the script:

`git diff -w --no-color | git apply --cached --ignore-whitespace && git checkout -- . && git reset && git add -A`

The `update-user-guide-links.py` file should probably be removed before merging.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/3438)

### CODE REVIEW

**Comment by PaulJuliusMartinez** [comment]
Think I signed the CLA.

File: docs/concepts/services-networking/ingress.md [hunk 1]
```diff
@@ -273,7 +273,7 @@ You can achieve the same by invoking `kubectl replace -f` on a modified Ingress
 
 ## Failing across availability zones
 
-Techniques for spreading traffic across failure domains differs between cloud providers. Please check the documentation of the relevant Ingress controller for details. Please refer to the federation [doc](/docs/user-guide/federation/) for details on deploying Ingress in a federated cluster.
+Techniques for spreading traffic across failure domains differs between cloud providers. Please check the documentation of the relevant Ingress controller for details. Please refer to the federation [doc](/docs/concepts/cluster-administration/federation.md) for details on deploying Ingress in a federated cluster.
```
Comment (chenopis) [inline docs/concepts/services-networking/ingress.md]: To preempt the redirect, you should leave off the `.md` and end it w/ a forward-slash, so `/docs/concepts/cluster-administration/federation/`.

File: docs/update-user-guide-links.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
+import subprocess
+import re
+
```
Comment (chenopis) [inline docs/update-user-guide-links.py]: Since this isn't run w/ every build, could we put it in a _tools or _scripts directory?

**Comment by chenopis** [comment]
Actually, I'm going to merge this now and address those 2 issues in a following PR so that we don't run into merge conflicts since this spans so many files.

**Comment by PaulJuliusMartinez** [comment]
Cool, thanks for accepting, @chenopis!

The `.md` link you issued actually stems from the [`docs/user-guide/federation/index.md`](https://raw.githubusercontent.com/kubernetes/kubernetes.github.io/master/docs/user-guide/federation/index.md) file, which says the content has been moved to `/docs/concepts/cluster-administration/federation.md`.

Also, I was just planning on removing the script before merging, but I suppose it might be useful in the future if content ever gets moved around again.

**Comment by chenopis** [comment]
Oh man, that link was actually my mistake then since I moved that federation content. :(

Would it be possible for you to generalize the script in a different PR so that it can take an input for the include filename to search for? We definitely are moving other sections around besides just user-guide.

**Pull request merged**